gocd/rabbit-openqa.py

```diff
@@ -100,12 +101,38 @@ def fetch_openqa_jobs(self, staging, iso, openqa_infos):
         # collect job infos to pick names
         for job in openqa:
             print(staging, iso, job['id'], job['state'], job['result'],
-                  job['settings']['MACHINE'], job['settings']['TEST'])
+                  job['settings']['FLAVOR'], job['settings']['TEST'], job['settings']['MACHINE'])
             openqa_infos[job['id']] = {'url': self.listener.test_url(job)}
             openqa_infos[job['id']]['state'] = self.map_openqa_result(job)
+            openqa_infos[job['id']]['flavor'] = job['settings']['FLAVOR']
             openqa_infos[job['id']]['name'] = job['settings']['TEST']
             openqa_infos[job['id']]['machine'] = job['settings']['MACHINE']
 
+    def check_name_combinations(openqa_info):
+        return [f"{openqa_info['name']}",
+                f"{openqa_info['name']}@{openqa_info['machine']}",
+                f"{openqa_info['flavor']}-{openqa_info['name']}@{openqa_info['machine']}"]
+
+    def make_unique_check_names(openqa_infos):
+        # Make a map of "possible name" -> [jobs]
+        map_name_jobs = defaultdict(list)
+        for id in openqa_infos:
+            for name in Project.check_name_combinations(openqa_infos[id]):
+                map_name_jobs[name].append(id)
+
+        # For each job, pick the shortest unique name
```

Comment:
when you're talking about stable output, consider the situation that your event comes in after the first flavor was scheduled and then another event comes in after the second flavor. Will you still name the jobs the same?

Reply:
I also thought about that and it's basically what I'd like to fix with #2894.

Reply:
Just to document what I discussed with Fabian in RL: I would suggest to have those names we know conflicting in its full spelling as required check. And if there is a required check matching a candidate name, pick it.

Reply:
To document my RL reply: I don't like that approach either, because it's not obvious why it's necessary, it's easy to forget to add a check on (seemingly simple) changes on the openQA side, and when (not if!) something goes wrong the failure mode is really unintuitive.